krm-functions/nad-fn/README.md

```diff
@@ -0,0 +1,116 @@
+
+Let's see an example! Assuming that a kpt package contains the following three resources (among others):
+
+    apiVersion: nf.nephio.org/v1alpha1
+    kind: UPFDeployment
+    metadata:
+        creationTimestamp: "Fri Mar 31 05:26:10 PM IST 2023"
+        name: upf-us-central1
+    spec:
+        capacity:
+            downlinkThroughput: 10G
+            uplinkThroughput: 1G
+        n3Interfaces:
+        - gatewayIPs:
+            - ""
+            ips:
+            - ""
+            name: n3
+    status:
+        computeuptime: "Fri Mar 31 05:26:10 PM IST 2023"
+        operationuptime: "Fri Mar 31 05:26:10 PM IST 2023"
+    ---
+    apiVersion: infra.nephio.io/v1alpha1
+    kind: ClusterContext
+    metadata:
+        name: cluster-context-1
+        annotations:
+            config.kubernetes.io/local-config: "true"
+    spec:
+        cniConfig:
+            cniType: "macvlan"
+            masterInterface: "bond0"
+        siteCode: edge1
+        region: us-central1
+    ---
+    apiVersion: ipam.nephio.org/v1alpha1
+    kind: IPAllocation
+    metadata:
+        creationTimestamp: null
+        labels:
+            nephio.org/interface: n3
+        name: upf-us-central1-n3
+    spec:
+        kind: network
+        selector:
+            matchLabels:
+            nephio.org/network-instance: sample-vpc
+            nephio.org/network-name: sample-n3-net
+    status:
+        prefix: 10.0.0.3/24
+        gateway: "10.0.0.1"
+
+the following list of CRs will be added to package:
+
+    apiVersion: "k8s.cni.cncf.io/v1"
+    kind: NetworkAttachmentDefinition
+    metadata:
+        name: upf-us-central1-n3
+    spec:
+        config: '{ "cniVersion": "0.3.1", "plugins": [ { "type": "macvlan", "capabilities": { "ips": true }, "master": "bond0", "mode": "bridge", "ipam": { "type": "static", "addresses": [ { "address": "10.0.0.3/24", "gateway": ""10.0.0.1"" } ] } }, { "capabilities": { "mac": true }, "type": "tuning" } ] }'
+    ---
+
+The function also adds the following conditions to the `Status` of the `Kptfile` object:
+
+    apiVersion: kpt.dev/v1
+    kind: Kptfile
+    metadata:
+        name: pkg-upf
+        annotations:
+            config.kubernetes.io/local-config: "true"
+    pipeline:
+      mutators:
+      - image: nephio.org/interface-manager:v1
+      - image: nephio.org/vlan:v1
+      - image: nephio.org/nad-fn:v1
+      - .....
+    status:
+        conditions:
+            - type: k8s-cni-cncf-io-v1-NetworkAttachmentDefinition-upf-us-central1-n3
+              status: true
+
+
+Implementation details
+----------------------
+
+The function ensures that the resources of an `NetworkAttachmentDefinition` are always in sync with the `Spec` of the input resources: the `ClusterContext`, `Network Function Deployments` and `IPAllocation`. In addition, it also checks for any previous `NetworkAttachmentDefinition` that exist part of kpt package and act upon them to produce the latest on every run. 
+
```

Comment:
Once the design details and client APIs for the function runtime library is confirmed , it would be good to add a few points here on how that library will be utilized for this function in implementation notes.

Reply:
WIll revisit this and add more context.